src/deploy/functions/cache/applyHash.ts

```diff
@@ -0,0 +1,57 @@
+import { Backend, Endpoint, allEndpoints } from "../backend";
+import * as args from "../args";
+import {
+  getEndpointHash,
+  getEnvironmentVariablesHash,
+  getSecretsHash,
+  getSourceHash,
+} from "./hash";
+
+/**
+ *
+ * Updates all the CodeBase {@link Backend}, applying a hash to each of their {@link Endpoint}.
+ */
+export async function applyBackendHashToBackends(
+  wantBackends: Record<string, Backend>,
+  context: args.Context
+): Promise<void> {
+  // envHash
+  const envHash = getEnvHash(wantBackends);
+  for (const [codebase, wantBackend] of Object.entries(wantBackends)) {
+    const source = context?.sources?.[codebase]; // populated earlier in prepare flow
+    const sourceV1Hash = source?.functionsSourceV1
+      ? await getSourceHash(source?.functionsSourceV1)
+      : undefined;
+    const sourceV2Hash = source?.functionsSourceV2
+      ? await getSourceHash(source?.functionsSourceV2)
+      : undefined;
+    applyBackendHashToBackend(wantBackend, envHash, sourceV1Hash, sourceV2Hash);
+  }
+}
+
+/**
+ * Updates {@link Backend}, applying a unique hash to each {@link Endpoint}.
+ */
+export function applyBackendHashToBackend(
+  wantBackend: Backend,
+  envHash: string,
+  sourceV1Hash?: string,
+  sourceV2Hash?: string
+): void {
+  allEndpoints(wantBackend).forEach((endpoint: Endpoint) => {
+    const secretsHash = getSecretsHash(endpoint);
+    const isV2 = endpoint.platform === "gcfv2";
+    const sourceHash = isV2 ? sourceV2Hash : sourceV1Hash;
+    // Blocked by #4866
```

Comment:
Is this still blocked? If so, maybe we should make this PR merge into that one and remove this comment

Reply:
Ah it got merged, no longer blocked, thanks for catching!!